### PR TITLE
Pierre b patch 1

### PR DESCRIPTION
**Pull request by PierreBernardCSMI**
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes?
- [ ] Have you successfully run the Feel++ testsuite with your changes locally?
- [ ] Have you written Doxygen comments in your contribution ?

-----

### CODE REVIEW

**Comment by PierreBernardCSMI** [comment]
I can't properly declare any function like this :``` template <typename T>
std::optional<holo3_image<T>>

**Comment by prudhomm** [comment]
The type is missing for holo3_image.

What is the error message?
Do you manage to compile now?

**Comment by prudhomm** [comment]
The type is missing for holo3_image.

What is the error message?
Do you mana to compile now?

**Comment by PierreBernardCSMI** [comment]
> What is the error message?

Sorry, i just edited my first comment with the error message in it

> Do you manage to compile now?

Yeah, thanks again it's working properly

**Comment by prudhomm** [comment]
But this is for holo3 repo Study, no?
Not feelpp

File: feelpp/feel/feelfilters/hbf.cpp [hunk 1]
```diff
@@ -543,7 +542,6 @@ template void writeHBF( std::string const& , holo3_image<int8_t> const& );
 template void writeHBF( std::string const& , holo3_image<int16_t> const& );
 template void writeHBF( std::string const& , holo3_image<int32_t> const& );
 template void writeHBF( std::string const& , holo3_image<int64_t> const& );
-template void writeHBF( std::string const& , holo3_image<long> const& );
```
Comment (PierreBernardCSMI) [inline feelpp/feel/feelfilters/hbf.cpp]: long type is the same as int64_t or int32_t depending on the compiler. In both cases, the specialization for long type was useless and was causing errors

**Comment by prudhomm** [comment]
Regarding optional can you show me how you calk the function?

**Comment by PierreBernardCSMI** [comment]
Yes, you are right, this might not be the perfect place to talk about this problem, sorry

**Comment by PierreBernardCSMI** [comment]
> Regarding optional can you show me how you call the function?

```c++
template <typename T>
std::optional<holo3_image<T>> Case::px() const
{
    if ( hasSlopes() )
    {
        return readHBF( ( M_case_p / "2.hbf" ).string() );
    }
    return std::nullopt;
}
```

**Comment by prudhomm** [comment]
See my comments on the holo3 repo

**Pull request merged**